pkg/crd/osm/v1/operating_system_profile.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	// OperatingSystemProfileResourceName represents "Resource" defined in Kubernetes
+	OperatingSystemProfileResourceName = "operatingsystemprofile"
+
+	// OperatingSystemProfileKindName represents "Kind" defined in Kubernetes
+	OperatingSystemProfileKindName = "OperatingSystemProfile"
+)
+
+//+genclient
+//+genclient:Namespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OperatingSystemProfile is the object that represents the OperatingSystemProfile
+type OperatingSystemProfile struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// OperatingSystemProfileSpec represents the operating system confuration spec.
+	Spec OperatingSystemProfileSpec `json:"spec"`
+}
+
+// OperatingSystemProfileSpec represents the data in the newly created OperatingSystemProfile
+type OperatingSystemProfileSpec struct {
+	// OSType represent the operating system name e.g: ubuntu
+	OSName string `json:"osName"`
+	// OSVersion the version of the operaitng system
+	OSVersion string `json:"osVersion"`
+	// SupportedCloudProviders represent the cloud providers that support the os name and version
```

Comment:
Nit: it might sound better to say something along
```suggestion
	// SupportedCloudProviders represent the cloud providers that support the given operating system version
```